### DISCUSSION TITLE
Establishing Relationships in a Meta-Model: Troubleshooting .odesign File Errors

### DISCUSSION THREAD

**Discussion by imeneazouzi** [eclipse-sirius/sirius-desktop#430] [Q&A]
While examining the meta-model (Fig. 1), I want to establish a relationship between the different flows. I define this in an .odesign file, as shown in the images above (Figs. 2 and 3), but it returns the error displayed in Fig. 4. Is it not possible to create relationships using the  reference, and if so, what should I correct to make it work?
**Fig1**
![Annotation 2024-07-02 173448](https://github.com/eclipse-sirius/sirius-desktop/assets/106630138/1d34af40-ba1f-4d3c-a5a3-2f3ec1b54681)
**Fig2**
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/106630138/d73635b2-9db5-498a-9b5b-4917e3cd7379)
**Fig3**
Here, I have defined a tool section in the .odesign file where I handle the creation of new edge for the class "Relation".
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/106630138/108d6cb6-f03e-4eb5-81d8-5b38d329954a)
_________________________________________________________________
Here, I have added the specification for the source and target.
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/106630138/0aa6b953-f489-47c9-a409-8a230c7157ca)

**Fig4**
This error message suggests that there is an attempt to access a feature named relationf on an instance of ProductFlowImpl within your meta_model. However, this feature (relationf) does  exist ;
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/106630138/b55f1851-4320-4f65-9070-ecf64b027566)

**Comment by lredor** [comment]
Hello,
I'm not sure, but I think that the expression `aql:self`, in the ChangeContext statement, is wrong. It returns a `ProductFlow`. You should have an expression that return the `Domain` where the new Relation should be created in.
Regards,
Laurent
